### PR TITLE
Merge 1.7.7.x to develop 24/02/21 [merge]

### PR DESCRIPTION
**Pull request by PierreRambaud**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Merge 177x to develop 24/02/21
| Type?             | improvement
| Category?         | ME
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | ~
| How to test?      | GA tests green and god bless us
| Possible impacts? | ~

CONFLICT (content): Merge conflict in tests/UI/pages/BO/catalog/products/add.js
CONFLICT (content): Merge conflict in tests/UI/campaigns/functional/BO/14_advancedParameters/08_logs/01_filterSortAndPagination.js
CONFLICT (content): Merge conflict in tests/Integration/Behaviour/Features/Context/Domain/OrderPaymentFeatureContext.php
CONFLICT (content): Merge conflict in src/Core/Domain/Order/Exception/OrderConstraintException.php
CONFLICT (content): Merge conflict in src/Adapter/Order/QueryHandler/GetOrderPreviewHandler.php
CONFLICT (content): Merge conflict in src/Adapter/Order/QueryHandler/GetOrderForViewingHandler.php
CONFLICT (content): Merge conflict in composer.lock
CONFLICT (content): Merge conflict in composer.json
CONFLICT (content): Merge conflict in classes/Product.php
CONFLICT (content): Merge conflict in admin-dev/themes/new-theme/js/pages/order/create/product-renderer.js
CONFLICT (content): Merge conflict in admin-dev/themes/new-theme/js/pages/order/create/product-manager.js
Automatic merge failed; fix conflicts and then commit the result.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/23380)

### CODE REVIEW

**Review by matks — COMMENTED** [review]
Seems OK but linter disagrees

File: themes/classic/modules/ps_shoppingcart/modal.tpl [hunk 1]
```diff
@@ -59,9 +59,9 @@
                 <p class="product-price">{$product.price}</p>
```
Comment (Progi1984) [inline themes/classic/modules/ps_shoppingcart/modal.tpl:59]: @PrestaShop/qa-automation Are these modifications copied to the ps_shopppingcart module ?

File: src/Core/Domain/Order/Exception/OrderConstraintException.php [hunk 2]
```diff
@@ -39,5 +39,5 @@ class OrderConstraintException extends OrderException
     /**
      * @var int Is used when invalid (not string) internal note provided
      */
-    public const INVALID_INTERNAL_NOTE = 2;
+    public const INVALID_PAYMENT_METHOD = 2;
```
Comment (jolelievre) [inline src/Core/Domain/Order/Exception/OrderConstraintException.php]: Nope, both are needed! I think `INVALID_PAYMENT_METHOD` was added in 177 recently but `INVALID_INTERNAL_NOTE` has been in 178 for a while

**Review by jolelievre — DISMISSED** [review]

File: themes/classic/modules/ps_shoppingcart/modal.tpl (hunk 1 above)
Comment (boubkerbribri) [reply themes/classic/modules/ps_shoppingcart/modal.tpl:59]: No it's not, should we do it ?

File: src/Core/Domain/Order/Exception/OrderConstraintException.php (hunk 2 above)
Comment (PierreRambaud) [reply src/Core/Domain/Order/Exception/OrderConstraintException.php]: :cry:

**Review by Progi1984 — DISMISSED** [review]

**Review by matks — APPROVED** [review]

File: themes/classic/modules/ps_shoppingcart/modal.tpl (hunk 1 above)
Comment (Progi1984) [reply themes/classic/modules/ps_shoppingcart/modal.tpl:59]: Yes, please

**Review by Progi1984 — APPROVED** [review]

**Comment by matks** [comment]
Thank you @PierreRambaud

**Pull request merged**